### PR TITLE
KafkaSource make consumergroup optional, set to uuid if not specified

### PR DESCRIPTION
**Pull request by lberk**
This fixes knative/eventing-contrib#1056

### CODE REVIEW

**Comment by duglin** [comment]
thanks

**Comment by matzew** [comment]
Applying something like:

```
apiVersion: sources.knative.dev/v1alpha1
kind: KafkaSource
metadata:
  name: kafka-source
spec:
  bootstrapServers: my-cluster-kafka-bootstrap.kafka:9092
  topics: my-topic
  sink:
    apiVersion: serving.knative.dev/v1alpha1
    kind: Service
    name: event-display
```

I am getting:

```
k apply -f 020-kafka-source.yml 
Error from server (BadRequest): error when creating "020-kafka-source.yml": admission webhook "defaulting.webhook.kafka.sources.knative.dev" denied the request: mutation failed: cannot decode incoming new object: json: unknown field "apiVersion"
```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_defaults.go [hunk 1]
```diff
@@ -18,9 +18,13 @@ package v1alpha1
...
+
+	"github.com/google/uuid"
 )
 
 // SetDefaults ensures KafkaSource reflects the default values.
-func (r *KafkaSource) SetDefaults(ctx context.Context) {
-	// No defaults
+func (k *KafkaSource) SetDefaults(ctx context.Context) {
+	if k != nil && k.Spec.ConsumerGroup == "" {
+		k.Spec.ConsumerGroup = uuid.New().String()
```
Comment (matzew) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_defaults.go:28]: perhaps with some constant, as a prefix: https://github.com/knative/eventing-contrib/issues/1056#issuecomment-610425303  ?

**Comment by matzew** [comment]
related: https://github.com/knative/eventing-contrib/issues/1143

**Comment by matzew** [comment]
/lgtm
/hold

if you do not want to address the prefix, cancel the hold

**Comment by matzew** [comment]
when you check the actual consumer groups, on the cluster (via ` k -n kafka  exec -it my-cluster-kafka-0 -- bin/kafka-consumer-groups.sh --bootstrap-server localhost:9092 --list`), it currently lists the UUID (as expected).

perhaps some prefix might be nice :man_shrugging:

**Comment by lberk** [comment]
@matzew ok, updated with a prefix of `knative-kafka-source` and tests updated as well

**Comment by lberk** [comment]
flake
/test pull-knative-eventing-contrib-integration-tests

**Pull request merged**